esp-hal/src/dma/buffers.rs

```diff
@@ -1562,3 +1564,364 @@ impl DerefMut for DmaLoopBuf {
         self.buffer
     }
 }
+
+/// A Preparation that masks itself as a DMA buffer.
+///
+/// Fow low level use, where none of the pre-made buffers really fit.
+///
+/// This type likely never should be visible outside of esp-hal.
+pub(crate) struct NoBuffer(Preparation);
+impl NoBuffer {
+    fn prep(&self) -> Preparation {
+        Preparation {
+            start: self.0.start,
+            direction: self.0.direction,
+            #[cfg(psram_dma)]
+            accesses_psram: self.0.accesses_psram,
+            burst_transfer: self.0.burst_transfer,
+            check_owner: self.0.check_owner,
+            auto_write_back: self.0.auto_write_back,
+        }
+    }
+}
+unsafe impl DmaTxBuffer for NoBuffer {
+    type View = Self;
+
+    fn prepare(&mut self) -> Preparation {
+        self.prep()
+    }
+
+    fn into_view(self) -> Self::View {
+        self
+    }
+
+    fn from_view(view: Self::View) -> Self {
+        view
+    }
+}
+unsafe impl DmaRxBuffer for NoBuffer {
+    type View = Self;
+
+    fn prepare(&mut self) -> Preparation {
+        self.prep()
+    }
+
+    fn into_view(self) -> Self::View {
+        self
+    }
+
+    fn from_view(view: Self::View) -> Self {
+        view
+    }
+}
+
+/// Prepares data unsafely to be transmitted via DMA.
+///
+/// `block_size` is the requirement imposed by the peripheral that receives the data. It
+/// ensures that the DMA will not try to copy a partial block, which would cause the RX DMA (that
+/// moves results back into RAM) to never complete.
+///
+/// The function returns the DMA buffer, and the number of bytes that will be transferred.
+///
+/// # Safety
+///
+/// The caller must keep all its descriptors and the buffers they
+/// point to valid while the buffer is being transferred.
+#[cfg_attr(not(aes_dma), expect(unused))]
+pub(crate) unsafe fn prepare_for_tx(
```

Comment:
For the sake of scoping (and auto complete in IDEs 😅), I think the bulk of the functions below should go into `impl NoBuffer { /* here */ }` if they are to live in this file/module.